amadeus-streaming/src/proto_util.rs

```diff
@@ -0,0 +1,144 @@
+const WIRE_TYPE_VAR_INT: u32 = 0;
+const PRECISION_OR_NUM_BUCKETS_FIELD_NUMBER: u32 = 3;
+pub const PRECISION_OR_NUM_BUCKETS_TAG: u32 =
+	PRECISION_OR_NUM_BUCKETS_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+
+const DATA_FIELD_NUMBER: u32 = 5;
+const WIRE_TYPE_LENGTH_DELIMITED: u32 = 2;
+pub const DATA_TAG: u32 = DATA_FIELD_NUMBER << 3 | WIRE_TYPE_LENGTH_DELIMITED;
+
+const AGGREGATOR_STATE_PROTO_TYPE_FIELD_NUMBER: u32 = 1;
+pub const TYPE_TAG: u32 = AGGREGATOR_STATE_PROTO_TYPE_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+
+const AGGREGATOR_STATE_PROTO_NUM_VALUES_FIELD_NUMBER: u32 = 2;
+pub const AGGREGATION_TYPE: i32 = 112;
+pub const NUM_VALUES_TAG: u32 =
+	AGGREGATOR_STATE_PROTO_NUM_VALUES_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+
+const AGGREGATOR_STATE_PROTO_ENCODING_VERSION_FIELD_NUMBER: u32 = 3;
+pub const ENCODING_VERSION_TAG: u32 =
+	AGGREGATOR_STATE_PROTO_ENCODING_VERSION_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+
+const DEFAULT_ENCODING_VERSION: u32 = 1;
+pub const HYPERLOGLOG_PLUS_PLUS_ENCODING_VERSION: i32 = 2;
+
+const AGGREGATOR_STATE_PROTO_VALUE_TYPE_FIELD_NUMBER: u32 = 4;
+pub const VALUE_TYPE_TAG: u32 =
+	AGGREGATOR_STATE_PROTO_VALUE_TYPE_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+pub const BYTES_OR_UTF8_STRING_TYPE: i32 = 11;
+
+const AGGREGATOR_STATE_PROTO_HYPERLOGLOG_PLUS_UNIQUE_STATE_FIELD_NUMBER: u32 = 112;
+pub const HYPERLOGLOGPLUS_UNIQUE_STATE_TAG: u32 =
+	AGGREGATOR_STATE_PROTO_HYPERLOGLOG_PLUS_UNIQUE_STATE_FIELD_NUMBER << 3
+		| WIRE_TYPE_LENGTH_DELIMITED;
+
+const SPARSE_PRECISION_OR_NUM_BUCKETS_FIELD_NUMBER: u32 = 4;
+pub const SPARSE_PRECISION_OR_NUM_BUCKETS_TAG: u32 =
+	SPARSE_PRECISION_OR_NUM_BUCKETS_FIELD_NUMBER << 3 | WIRE_TYPE_VAR_INT;
+
+const MAX_VAR_INT_SIZE: usize = 10;
+
+pub fn compute_uint32_size_no_tag(value: u32) -> usize {
+	if value & (!0 << 7) == 0 {
+		return 1;
+	}
+	if value & (!0 << 14) == 0 {
+		return 2;
+	}
+	if value & (!0 << 21) == 0 {
+		return 3;
+	}
+	if value & (!0 << 28) == 0 {
+		return 4;
+	}
+	return 5;
```

Comment:
```suggestion
	5
```
Minor thing but this is triggering `clippy`

Reply:
Fixed